File: src/code/cpp/binary_search/binary_search.cpp
#include <vector>

using namespace std;


int fn(vector<int>& arr, int target) {
    int left = 0;
    int right = arr.size() - 1;

    while (left <= right) {
        int mid = left + (right - left) / 2;

        if (arr[mid] == target) {
            // TODO: logic
            return;
        }
        if (arr[mid] > target) {
            right = mid - 1;
        } else {
            left = mid + 1;
        }
    }

    return left;
}

File: src/code/ruby/binary_search/binary_search.rb
def fn(arr, target)
  left  = 0
  right = arr.length - 1

  while left <= right
    mid = (left + right) / 2

    if arr[mid] == target
      # TODO: logic
      return
    elsif arr[mid] > target
      right = mid - 1
    else
      left = mid + 1
    end
  end

  left
end
